Cargo.toml

```diff
@@ -35,6 +35,8 @@ members = [
 ]
 
 [workspace.dependencies]
+async-trait = { version = "0.1.77" }
```

Comment:
Do we use dynamic dispatch with async traits anywhere? We could replace any traits that don't need to be dyn with native async traits now (or rather native impl trait in traits because we'd need to specify the send bound on the resulting future because of tokio; the implementations of those traits can use async functions though) and get the performance benefit of not boxing the futures.

Reply:
There's now also an official proc macro from the Rust project to automate adding those `Send` bounds to native async traits, but it creates a new trait while also keeping the original non-send one: https://crates.io/crates/trait-variant

Reply:
(out of scope of this PR of course, just thought of it when I saw `async-trait` in diff)

Reply:
Thanks Alexey! I can address this in another PR.

Reply:
We do, `Connector` and `Transaction` traits in quaint at least